### PR TITLE
Move offices model to tenants

### PR DESCRIPTION
**Pull request by reinbach**
[ch6742]

### CODE REVIEW

File: src/etools/applications/reports/migrations/0021_auto_20191011_1201.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Generated by Django 2.2.4 on 2019-10-11 12:01
+
+from django.db import migrations
+
+
+def copy_offices_data(apps, schema):
+    OfficeOld = apps.get_model("users", "office")
+    Office = apps.get_model("reports", "office")
+    for old in OfficeOld.objects.all():
+        Office.objects.create(
```
Comment (domdinicola) [inline src/etools/applications/reports/migrations/0021_auto_20191011_1201.py]: question: does this manager filter only offices from the current tenant?

File: src/etools/applications/reports/models.py [hunk 2]
```diff
@@ -850,3 +853,43 @@ class Meta:
...
+    )
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        ordering = ('name', )
+
+
+class UserProfileOffice(models.Model):
```
Comment (domdinicola) [inline src/etools/applications/reports/models.py]: maybe we should call it TenantProfile (one day we may want to add other things which are related to the tenant e.g. sections)

File: src/etools/applications/reports/migrations/0021_auto_20191011_1201.py (hunk 1 above)
Comment (reinbach) [reply src/etools/applications/reports/migrations/0021_auto_20191011_1201.py]: It should if the `connection.tenant` is being set during migration

File: src/etools/applications/reports/models.py (hunk 2 above)
Comment (reinbach) [reply src/etools/applications/reports/models.py]: Changed it to `UserTenantProfile`

File: src/etools/applications/reports/migrations/0021_auto_20191011_1201.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Generated by Django 2.2.4 on 2019-10-11 12:01
+
+from django.db import migrations
+
+
+def copy_offices_data(apps, schema):
+    OfficeOld = apps.get_model("users", "office")
```
Comment (robertavram) [inline src/etools/applications/reports/migrations/0021_auto_20191011_1201.py]: the manager here will not work, add below logic in this migration

File: src/etools/applications/reports/views/v2.py [hunk 4]
```diff
@@ -491,3 +496,33 @@ def finalize_response(self, request, response, *args, **kwargs):
                 intervention.reference_number
             )
         return response
+
```
Comment (robertavram) [inline src/etools/applications/reports/views/v2.py]: not related to this line but making a note that we should check if all old relationships are preserved well.

File: src/etools/applications/users/migrations/0013_auto_20191010_1621.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('users', '0012_auto_20190513_1804'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='office',
+            name='zonal_chief',
+            field=models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='offices_old', to=settings.AUTH_USER_MODEL, verbose_name='Chief'),
```
Comment (robertavram) [inline src/etools/applications/users/migrations/0013_auto_20191010_1621.py]: we can get rid of this field.

File: src/etools/applications/reports/models.py [hunk 6]
```diff
@@ -850,3 +853,43 @@ class Meta:
...
+    class Meta:
+        ordering = ('name', )
+
+
+class UserTenantProfile(models.Model):
+    profile = models.OneToOneField(
+        UserProfile,
+        verbose_name=_('Profile'),
+        on_delete=models.CASCADE,
+        related_name="profile_office",
```
[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/reports/models.py]: done

File: src/etools/applications/reports/migrations/0021_auto_20191011_1201.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    Office = apps.get_model("reports", "office")
+    for old in OfficeOld.objects.all():
+        Office.objects.create(
+            pk=old.pk,
+            name=old.name,
+            zonal_chief=old.zonal_chief,
```
[the remark this answers is not in the document]
Comment (reinbach) [reply src/etools/applications/reports/migrations/0021_auto_20191011_1201.py]: removed

File: src/etools/applications/reports/migrations/0021_auto_20191011_1201.py (hunk 3 above)
Comment (reinbach) [reply src/etools/applications/reports/migrations/0021_auto_20191011_1201.py]: done

**Comment by reinbach** [comment]
@robertavram Locally I'm running into an issue where a t2f.Travel record (pk: 33) in `myanmar` schema references an office record that is not specific to a country. This is causing a migration to fail, as that office record is not being migrated in the data migration to the schema.

Office record is `{'pk': 14, 'name': 'Sittwe'}` it exists in Dev environment without a relation to a schema either, but there is not a similar t2f.Travel record that references this office. I'm not sure about Staging or Production env.

If this is just an issue locally, thing I think things look good here

**Comment by domdinicola** [comment]
@reinbach can you create an Admin for the UserTenantProfile model

File: src/etools/applications/users/migrations/0013_auto_20191010_1621.py (hunk 5 above)
Comment (domdinicola) [reply src/etools/applications/users/migrations/0013_auto_20191010_1621.py]: migration has been removed

File: src/etools/applications/reports/views/v2.py (hunk 4 above)
Comment (reinbach) [reply src/etools/applications/reports/views/v2.py]: Old pk is copied across to new table, maintaining relationship with objects/models

**Pull request merged**